drivers/xen/gnttab.c

```diff
@@ -306,8 +311,18 @@ static int gnttab_init(const struct device *d)
 		put_free_entry(gref);
 	}
 
-	gnttab.table = (grant_entry_v1_t *)
-			DT_REG_ADDR_BY_IDX(DT_INST(0, xen_xen), 0);
+	/*
+	 * User to internal flag conversions:
+	 * K_MEM_CACHE_WB  --> MT_NORMAL
+	 * K_MEM_PERM_RW   --> MT_RW
+	 * K_MEM_PERM_EXEC --> MT_P_EXECUTE_NEVER
+	 *
+	 * note: arm64 __arch_mem_map func also sets the next flags by default:
+	 *       * MT_DEFAULT_SECURE_STATE
+	 *       * MT_P_RX_U_NA
+	 */
+	DEVICE_MMIO_TOPLEVEL_MAP(grant_tables, K_MEM_CACHE_WB | K_MEM_PERM_RW | K_MEM_PERM_EXEC);
```

Comment:
This region should not be executable:

> #define K_MEM_PERM_EXEC   BIT(4)
Region will be executable (normally forbidden)

 last flag should be removed.